### ISSUE TITLE
RUF075 false-positive with a context manager

### ISSUE THREAD

**Issue by eternal-sorrow** [astral-sh/ruff#27552]
### Summary

Example looks something like this:

```
@contextmanager
def some_func():
    with some_context_manager():
        yield
        some_code()
```

Cleanup here is handled by the context manager. `some_code()` here is not a cleanup code. It's supposed to run only if no exception was raised.

### Version

0.16.1

**Comment by ntBre** [comment]
This exact example doesn't seem to trigger the rule: https://play.ruff.rs/0596b322-7522-431a-8f1b-2333a3dba831

Is there more surrounding context in an example you're encountering in a real project?

**Comment by eternal-sorrow** [comment]
I updated the example. This one does trigger it.

**Comment by ntBre** [comment]
Ah okay, in that case this has roughly the same cause as #25564. It was an intentional decision on the PR adding the rule to flag these, but if it keeps coming up, maybe we should reconsider.
